### ISSUE TITLE
Make "max-warnings" a proper option

### ISSUE THREAD

**Issue by nottrobin** [sasstools/sass-lint#856]
The command-line tool provides the `--max-warnings` option. This is very useful for preventing a codebase from deteriorating and generating ever more warnings.

However, we can't use this feature for [vanilla-framework](https://github.com/ubuntudesign/vanilla-framework/) as we run sass-lint through [gulp-sass-lint](https://github.com/sasstools/gulp-sass-lint) in our `gulpfile.js` instead of using the CLI.

As far as I can tell there's no way to set `program.maxWarnings` through any other method. It would be most helpful to us if this could be an [option](https://github.com/sasstools/sass-lint/tree/develop/docs/options) in the same sense as `formatter` et al., so we can set it within the `.sass-lint.yml` config file as follows:

``` yaml
options:
  max-warnings: 25
```

**Comment by nottrobin** [comment]
I'll have a little crack at writing this myself. Although if I can't get the bulk of the way there tonight I'll probably abandon it. :stuck_out_tongue_closed_eyes:

**Comment by DanPurdy** [comment]
Hi @nottrobin,
The difference here is that the CLI just throws an exit code of 1 if it goes over a certain number of warnings, this wouldn't be the case with gulp as you need to still report on every warning/file and then hand it off downstream. Would you expect it to throw an extra error message if it went over the number of warnings you specify?
I'm not too clear on the use case here and what the intended outcome would be.

**Comment by nottrobin** [comment]
Thanks for the quick response @DanPurdy.

So currently, if I were to set `indentation: 2` in my `.sass-lint.yml` file and run `sass-lint`, I see an exception with an error message, and, as you say, an exit code of `1`:

``` bash
$ sass-lint
...
Error: 1 errors were detected in
- scss/_base_lists.scss
...
```

Similarly, if I have a `gulpfile.js` which includes the following:

``` javascript
var sassLint = require('gulp-sass-lint');

gulp.task('lint:sass', function() {
  return gulp.src('scss/**/*.s+(a|c)ss')
    .pipe(sassLint())
    .pipe(sassLint.format())
    .pipe(sassLint.failOnError())
});
```

And I run this target with gulp, I see an error, with an exit code of `1`.

``` bash
$ gulp lint:sass
...
[18:10:23] 'lint:sass' errored after 1.11 s
[18:10:23] Error in plugin 'sass-lint'
Message:
    1 errors detected in _base_lists.scss
```

We use `gulp` in our CI testing, and therefore rely on the exit code of `1` that it produces on an error.

I would expect very similar behaviour from a `max-warnings` option - that is that it generates an error message and an exit code of `1` whether it's run through the CLI or through `gulp`.

Currently if I run `sass-lint --max-warnings 10` I get an exit code of `1`, but no error message which is a shame. But crucially I'd like to be able to generate an exit code of `1` for the same reason through the `.sass-lint.yml` file.

I've had a go at developing this, in #857, so you can see what I mean, but it's not ready yet as I haven't made any tests.

**Comment by nottrobin** [comment]
I have just discovered that this won't solve my problem with gulp-sass-lint, as instead of using sass-lint's `failOnError` method, it [reimplements its own version of that function](https://github.com/sasstools/gulp-sass-lint/blob/master/index.js#L81).

Once this is merged, I'd be happy to submit a follow-up PR to gulp-sass-lint to make use of the native `failOnError` method and so properly support the `max-warnings` option.
